File: core/ImplementQueueUsingStacks/ImplementQueueUsingStacks.py
class MyQueue:
    def __init__(self):
        self.stack1 = []
        self.stack2 = []

    def push(self, x: int) -> None:
        while self.stack2:
            self.stack1.append(self.stack2.pop())
        self.stack1.append(x)

    def pop(self) -> int:
        while self.stack1:
            self.stack2.append(self.stack1.pop())
        return self.stack2.pop()

    def peek(self) -> int:
        while self.stack1:
            self.stack2.append(self.stack1.pop())
        return self.stack2[-1]

    def empty(self) -> bool:
        return len(self.stack1) == 0 and len(self.stack2) == 0

File: core/ImplementQueueUsingStacks/ImplementQueueUsingStacks.java
package com.dbc;

import java.util.Stack;

public class ImplementQueueUsingStacks {
    private final Stack<Integer> stack1 = new Stack<>();
    private final Stack<Integer> stack2 = new Stack<>();

    public ImplementQueueUsingStacks() {
    }

    public void push(int x) {
        while (!this.stack2.isEmpty()) this.stack1.push(this.stack2.pop());
        this.stack1.push(x);
    }

    public int pop() {
        while (!this.stack1.isEmpty()) this.stack2.push(this.stack1.pop());
        return this.stack2.pop();
    }

    public int peek() {
        while (!this.stack1.isEmpty()) this.stack2.push(this.stack1.pop());
        return this.stack2.peek();
    }

    public boolean empty() {
        return this.stack1.isEmpty() && this.stack2.isEmpty();
    }
}
